pkg/rehearse/jobs.go

```diff
@@ -222,6 +228,107 @@ func ConfigureRehearsalJobs(toBeRehearsed config.Presubmits, ciopConfigs config.
 	return rehearsals
 }
 
+// AddRandomJobsForChangedTemplates finds random jobs from the PR config that are using a specific template with a specific cluster type.
+// So if a template will be changed, find the jobs that are using a template in combination with the `aws`,`openstack`,`gcs` and `libvirt` cluster types.
+func AddRandomJobsForChangedTemplates(templatesWithClusterTypes map[string]sets.String, templates config.CiTemplates, prConfigPresubmits map[string][]prowconfig.Presubmit, loggers Loggers, prNumber int) config.Presubmits {
+	rehearsals := make(config.Presubmits)
+
+	for templateFile, clusterTypes := range templatesWithClusterTypes {
+		for clusterType := range clusterTypes {
+			if repo, job := pickTemplateJob(prConfigPresubmits, templateFile, clusterType); job != nil {
+				jobLogger := loggers.Job.WithFields(logrus.Fields{"target-repo": repo, "target-job": job.Name})
+
+				volumeRef, err := getVolumeIndexAndTempCMName(job.Spec.Containers[0].VolumeMounts, job.Spec.Volumes, templates)
+				if err != nil {
+					jobLogger.WithError(err).Warn(skipJobMessage)
+					continue
+				}
+
+				if volumeRef != nil {
+					job.Spec.Volumes[volumeRef.index].VolumeSource.ConfigMap.Name = volumeRef.name
+				}
+
+				rehearsals[repo] = append(rehearsals[repo], *job)
+			}
+		}
+	}
+	return rehearsals
+}
+
+type volumeRef struct {
+	index       int
+	name        string
+	templateKey string
+}
+
+func getVolumeIndexAndTempCMName(volumeMounts []v1.VolumeMount, volumes []v1.Volume, templates config.CiTemplates) (*volumeRef, error) {
+	exists, index, templateKey := hasChangedTemplateVolume(volumeMounts, volumes, templates)
+	if !exists {
+		return nil, nil
+	}
+
+	templateData, err := config.GetTemplateData(templates[templateKey])
+	if err != nil {
+		return nil, fmt.Errorf("couldn't get template's data: %s: %v", templates[templateKey], err)
+	}
+	return &volumeRef{
+		index:       index,
+		name:        config.GetTempCMName(config.GetTemplateName(templateKey), templateKey, templateData),
+		templateKey: templateKey}, nil
+}
+
+func getClusterTypeEnvIndex(envVars []v1.EnvVar) (bool, int) {
```

Comment:
Return -1 (as the stdlib does) as the index instead of making the return value `bool, int`.

Reply:
nice idea.